### ISSUE TITLE
kernel weight error from converting customized models (tensorRT)

### ISSUE THREAD

**Issue by youngjae-avikus** [wang-xinyu/tensorrtx#935]
## Env

- GPU, e.g. V100, RTX2080, TX2, Xavier NX, Nano, etc.
- OS, e.g. Ubuntu16.04, Win10, etc. `Xavier AGX`
- Cuda version `10.2`
- TensorRT version `7.1.3`

## About this repo

- which branch/tag/commit are you using?
- which model? yolov5, retinaface? yolov5m-v5

## Your problem

- what is your command? e.g. `sudo ./yolov5 -s`
- what's your output?
- what output do you expect?

Hello, I successfully adapted the deepstream h264 sample execution through this repo with coco pretrained weight(yolov5m.pt) made by yolov5-v5(ultralytics)

I'm going to test the custom model that I made before, but there was a kernel weight error. I've tuned personal model through evolve, and I think it's hyp:anchors that influence the model architecture, but it's actually unclear whether anchors affect the model architecture detect. Is not it simply an extension of the anchor when calculating the loss? Anyway, the summary of my model is as follows.

{'nc': 5, 'depth_multiple': 0.67, 'width_multiple': 0.75, 'anchors': 4, 'backbone': [[-1, 1, 'Focus', [64, 3]], [-1, 1, 'Conv', [128, 3, 2]], [-1, 3, 'C3', [128]], [-1, 1, 'Conv', [256, 3, 2]], [-1, 9, 'C3', [256]], [-1, 1, 'Conv', [512, 3, 2]], [-1, 9, 'C3', [512]], [-1, 1, 'Conv', [1024, 3, 2]], [-1, 1, 'SPP', [1024, [5, 9, 13]]], [-1, 3, 'C3', [1024, False]]], 'head': [[-1, 1, 'Conv', [512, 1, 1]], [-1, 1, 'nn.Upsample', ['None', 2, 'nearest']], [[-1, 6], 1, 'Concat', [1]], [-1, 3, 'C3', [512, False]], [-1, 1, 'Conv', [256, 1, 1]], [-1, 1, 'nn.Upsample', ['None', 2, 'nearest']], [[-1, 4], 1, 'Concat', [1]], [-1, 3, 'C3', [256, False]], [-1, 1, 'Conv', [256, 3, 2]], [[-1, 14], 1, 'Concat', [1]], [-1, 3, 'C3', [512, False]], [-1, 1, 'Conv', [512, 3, 2]], [[-1, 10], 1, 'Concat', [1]], [-1, 3, 'C3', [1024, False]], [[17, 20, 23], 1, 'Detect', ['nc', 'anchors']]], 'ch': 3}

Also, we revised the variables CLASS_NUM and CHECK_COUNT because they are anchors=4 and nc=5.
```cpp
namespace Yolo
{
    static constexpr int CHECK_COUNT = 4;
    static constexpr float IGNORE_THRESH = 0.1f;
    struct YoloKernel
    {
        int width;
        int height;
        float anchors[CHECK_COUNT * 2];
    };
    static constexpr int MAX_OUTPUT_BBOX_COUNT = 1000;
    static constexpr int CLASS_NUM = 5;
    static constexpr int INPUT_H = 640;  // yolov5's input height and width must be divisible by 32.
    static constexpr int INPUT_W = 640;

    static constexpr int LOCATIONS = 4;
    struct alignas(float) Detection {
        //center_x center_y w h
        float bbox[LOCATIONS];
        float conf;  // bbox_conf * cls_conf
        float class_id;
    };
}
```
The issue of this is constantly being talked about, but there is no exact solution. The depth_multiple and width_multiple variables are the same as the basic yolov5m.

Error
![image](https://user-images.githubusercontent.com/99321359/158183379-95fa8601-3ad9-4fa5-a901-852d58c06193.png)

What's the reason?
Thank you

**Comment by AsakusaRinne** [comment]
What is your settings on ```.yaml``` file when you trained your model? It seems the summary is a little different with that of ```.yaml``` file. I believe ```anchors``` is not supposed to influence the architecture, too.

For a possible way to figure out the mistake, you may use pretrained yolov5m weight as the input of tenssortx and print the "kernel weight count" of each layer. Thus you can compare the error message with them and try to locate the problem.

**Comment by yunxicdq** [comment]
Does your environment report this error when generating engine file?
This error is as follows:
![image](https://user-images.githubusercontent.com/101306803/158300426-448816c1-c402-4aea-b2f5-745c2276bfed.png)
